runtime/src/runtime/mod.rs

```diff
@@ -218,29 +224,53 @@ pub trait Runtime: Primitives + Verifier + RuntimePolicy {
     /// Returns the gas base fee (cost per unit) for the current epoch.
     fn base_fee(&self) -> TokenAmount;
 
+    /// The gas still available for computation
+    fn gas_available(&self) -> u64;
+
+    /// The current tipset's timestamp, as UNIX seconds
+    fn tipset_timestamp(&self) -> u64;
+
+    /// The hash of on of the last 256 blocks
+    fn tipset_cid(&self, epoch: i64) -> Option<Cid>;
```

Comment:
What happens if epoch is more than 256 ago? Just `None`? Kinda wierd we don't get error numbers here.

Reply:
Thanks, yes I would expect FINALITY too